### PR TITLE
Fixed interpreter support for properties on Nullable<T>

### PR DESCRIPTION
**Pull request by VSadov**
such as HasValue / Value

Also fixed a problem with invoking of GetValueOrDefault on null receiver discovered while testing.
We should return a default  value in such case, not the boxed receiver which is null and would cause NRE.

### CODE REVIEW

**Comment by VSadov** [comment]
@AlekseyTs - please review

File: src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs [hunk 1]
```diff
@@ -1899,7 +1899,7 @@ public override string ToString()
 
     internal abstract class NullableMethodCallInstruction : Instruction
     {
-        private static NullableMethodCallInstruction s_hasValue,s_value,s_equals,s_getHashCode,s_getValueOrDefault,s_getValueOrDefault1,s_toString;
+        private static NullableMethodCallInstruction s_hasValue,s_value,s_equals,s_getHashCode,s_getValueOrDefault1,s_toString;
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs]: Nit: I realize you only changed the name of one of these, but it'd be nice if there were spaces between each field.

File: src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs [hunk 2]
```diff
@@ -1940,9 +1940,20 @@ public override int Run(InterpretedFrame frame)
 
         private class GetValueOrDefault : NullableMethodCallInstruction
         {
+            private Type defaultValueType;
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs]: readonly?

File: src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs [hunk 3]
```diff
@@ -1940,9 +1940,20 @@ public override int Run(InterpretedFrame frame)
...
+            {
+                defaultValueType = mi.ReturnType;
+            }
+
             public override int Run(InterpretedFrame frame)
             {
-                // we're already unboxed in the interpreter, so this is a nop.
+                if (frame.Peek() == null)
+                {
+                        frame.Pop();
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs]: Nit: indentation looks off here

File: src/System.Linq.Expressions/tests/SequenceTests/SequenceTests.cs [hunk 4]
```diff
@@ -4266,6 +4266,31 @@ public static void TestConvertToNullable()
             Assert.Equal(2, d(2));
         }
 
+        [Fact]
```
Comment (stephentoub) [inline src/System.Linq.Expressions/tests/SequenceTests/SequenceTests.cs]: Remind me: will these tests run with both the compiled and interpreted versions?

File: src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs (hunk 1 above)
Comment (VSadov) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs]: no prob

File: src/System.Linq.Expressions/tests/SequenceTests/SequenceTests.cs (hunk 4 above)
Comment (VSadov) [inline src/System.Linq.Expressions/tests/SequenceTests/SequenceTests.cs]: I could not yet figure how to run tests in both modes. 
The problem is that interpretation is not a runtime switch, it requires rebuild. It is easy to run test in interpretation mode locally by setting an environment variable and forcing rebuild.

I will be porting these fixes to ProjectN branch where they will definitely run in interpreter mode, but yes, we need a way to run interpreter in CI.

Comment (VSadov) [inline src/System.Linq.Expressions/tests/SequenceTests/SequenceTests.cs]: Entered https://github.com/dotnet/corefx/issues/2756 to track this test hole

**Pull request merged**